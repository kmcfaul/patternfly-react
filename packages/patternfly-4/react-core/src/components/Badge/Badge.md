---
title: 'Badge'
cssPrefix: 'pf-c-badge'
---

import { Badge } from '@patternfly/react-core';

## Badge (read)

```js
import React from 'react';
import { Badge } from '@patternfly/react-core';

class ReadBadge extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Badge isRead>7</Badge>
        <Badge isRead>24</Badge>
        <Badge isRead>240</Badge>
        <Badge isRead>999+</Badge>
      </React.Fragment>
    );
  }
}
```

## Badge (unread)

```js
import React from 'react';
import { Badge } from '@patternfly/react-core';

class UnreadBadge extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Badge>7</Badge>
        <Badge>24</Badge>
        <Badge>240</Badge>
        <Badge>999+</Badge>
      </React.Fragment>
    );
  }
}
```
