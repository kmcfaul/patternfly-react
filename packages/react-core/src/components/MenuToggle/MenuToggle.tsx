import * as React from 'react';
import styles from '@patternfly/react-styles/css/components/MenuToggle/menu-toggle';
import { css } from '@patternfly/react-styles';
import CaretDownIcon from '@patternfly/react-icons/dist/js/icons/caret-down-icon';
import { BadgeProps } from '../Badge';

export interface MenuToggleProps
  extends React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> {
  /** Content rendered inside the toggle */
  children?: React.ReactNode;
  /** Additional classes added to the toggle */
  className?: string;
  /** Flag indicating the toggle has expanded styling */
  isExpanded?: boolean;
  /** Flag indicating the toggle is disabled */
  isDisabled?: boolean;
  /** Flag indicating the toggle should have primary styling */
  isPrimary?: boolean;
  /** Flag indicating the toggle should have plain styling and only render the children. Incompatible with icon and badge properties. Pass an icon via children to make an icon toggle. */
  isPlain?: boolean;
  /** Optional icon rendered inside the toggle, before the children content */
  icon?: React.ReactNode;
  /** Optional badge rendered inside the toggle, after the children content */
  badge?: BadgeProps | React.ReactNode;
}

export class MenuToggle extends React.Component<MenuToggleProps> {
  displayName = 'MenuToggle';
  static defaultProps = {
    className: '',
    isExpanded: false,
    isDisabled: false,
    isPrimary: false,
    isPlain: false
  };

  render() {
    const { children, className, icon, badge, isExpanded, isDisabled, isPrimary, isPlain, ...props } = this.props;

    const content = (
      <React.Fragment>
        {icon && <span className={css(styles.menuToggleIcon)}>{icon}</span>}
        <span className="pf-c-menu-toggle__text">{children}</span>
        {badge && <span className={css(styles.menuToggleCount)}>{badge}</span>}
        <span className={css(styles.menuToggleControls)}>
          <span className={css(styles.menuToggleToggleIcon)}>
            <CaretDownIcon aria-hidden />
          </span>
        </span>
      </React.Fragment>
    );
    return (
      <button
        className={css(
          styles.menuToggle,
          isExpanded && styles.modifiers.expanded,
          isPrimary && styles.modifiers.primary,
          isPlain && styles.modifiers.plain,
          className
        )}
        type="button"
        aria-expanded={false}
        {...(isExpanded && { 'aria-expanded': true })}
        {...(isDisabled && { disabled: true })}
        {...props}
      >
        {isPlain && children}
        {!isPlain && content}
      </button>
    );
  }
}
