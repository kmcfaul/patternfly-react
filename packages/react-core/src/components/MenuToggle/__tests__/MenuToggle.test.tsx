import React from 'react';
import { shallow } from 'enzyme';
import { MenuToggle } from '../MenuToggle';
import { Badge } from '../../Badge';
import CogIcon from '@patternfly/react-icons/dist/js/icons/cog-icon';
import EllipsisVIcon from '@patternfly/react-icons/dist/js/icons/ellipsis-v-icon';

describe('menu toggle', () => {
  test('renders successfully', () => {
    const view = shallow(<MenuToggle>Toggle</MenuToggle>);
    expect(view).toMatchSnapshot();
  });

  test('passes additional classes', () => {
    const view = shallow(<MenuToggle className="test-class">Toggle</MenuToggle>);
    expect(view).toMatchSnapshot();
  });

  test('shows isExpanded', () => {
    const view = shallow(<MenuToggle isExpanded>Toggle</MenuToggle>);
    expect(view).toMatchSnapshot();
  });

  test('shows isDisabled', () => {
    const view = shallow(<MenuToggle isDisabled>Toggle</MenuToggle>);
    expect(view).toMatchSnapshot();
  });

  test('shows isPrimary', () => {
    const view = shallow(<MenuToggle isPrimary>Toggle</MenuToggle>);
    expect(view).toMatchSnapshot();
  });

  test('shows isPlain', () => {
    const view = shallow(
      <MenuToggle isPlain>
        <EllipsisVIcon />
      </MenuToggle>
    );
    expect(view).toMatchSnapshot();
  });

  test('shows icon', () => {
    const view = shallow(<MenuToggle icon={<CogIcon />}>Toggle</MenuToggle>);
    expect(view).toMatchSnapshot();
  });

  test('shows badge', () => {
    const view = shallow(<MenuToggle badge={<Badge>badge</Badge>}>Toggle</MenuToggle>);
    expect(view).toMatchSnapshot();
  });
});
