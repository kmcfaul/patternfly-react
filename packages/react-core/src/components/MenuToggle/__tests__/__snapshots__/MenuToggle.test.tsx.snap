// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu toggle passes additional classes 1`] = `
<button
  aria-expanded={false}
  className="pf-c-menu-toggle test-class"
  type="button"
>
  <span
    className="pf-c-menu-toggle__text"
  >
    Toggle
  </span>
  <span
    className="pf-c-menu-toggle__controls"
  >
    <span
      className="pf-c-menu-toggle__toggle-icon"
    >
      <CaretDownIcon
        aria-hidden={true}
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
    </span>
  </span>
</button>
`;

exports[`menu toggle renders successfully 1`] = `
<button
  aria-expanded={false}
  className="pf-c-menu-toggle"
  type="button"
>
  <span
    className="pf-c-menu-toggle__text"
  >
    Toggle
  </span>
  <span
    className="pf-c-menu-toggle__controls"
  >
    <span
      className="pf-c-menu-toggle__toggle-icon"
    >
      <CaretDownIcon
        aria-hidden={true}
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
    </span>
  </span>
</button>
`;

exports[`menu toggle shows badge 1`] = `
<button
  aria-expanded={false}
  className="pf-c-menu-toggle"
  type="button"
>
  <span
    className="pf-c-menu-toggle__text"
  >
    Toggle
  </span>
  <span
    className="pf-c-menu-toggle__count"
  >
    <Badge>
      badge
    </Badge>
  </span>
  <span
    className="pf-c-menu-toggle__controls"
  >
    <span
      className="pf-c-menu-toggle__toggle-icon"
    >
      <CaretDownIcon
        aria-hidden={true}
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
    </span>
  </span>
</button>
`;

exports[`menu toggle shows icon 1`] = `
<button
  aria-expanded={false}
  className="pf-c-menu-toggle"
  type="button"
>
  <span
    className="pf-c-menu-toggle__icon"
  >
    <CogIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />
  </span>
  <span
    className="pf-c-menu-toggle__text"
  >
    Toggle
  </span>
  <span
    className="pf-c-menu-toggle__controls"
  >
    <span
      className="pf-c-menu-toggle__toggle-icon"
    >
      <CaretDownIcon
        aria-hidden={true}
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
    </span>
  </span>
</button>
`;

exports[`menu toggle shows isDisabled 1`] = `
<button
  aria-expanded={false}
  className="pf-c-menu-toggle"
  disabled={true}
  type="button"
>
  <span
    className="pf-c-menu-toggle__text"
  >
    Toggle
  </span>
  <span
    className="pf-c-menu-toggle__controls"
  >
    <span
      className="pf-c-menu-toggle__toggle-icon"
    >
      <CaretDownIcon
        aria-hidden={true}
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
    </span>
  </span>
</button>
`;

exports[`menu toggle shows isExpanded 1`] = `
<button
  aria-expanded={true}
  className="pf-c-menu-toggle pf-m-expanded"
  type="button"
>
  <span
    className="pf-c-menu-toggle__text"
  >
    Toggle
  </span>
  <span
    className="pf-c-menu-toggle__controls"
  >
    <span
      className="pf-c-menu-toggle__toggle-icon"
    >
      <CaretDownIcon
        aria-hidden={true}
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
    </span>
  </span>
</button>
`;

exports[`menu toggle shows isPlain 1`] = `
<button
  aria-expanded={false}
  className="pf-c-menu-toggle pf-m-plain"
  type="button"
>
  <EllipsisVIcon
    color="currentColor"
    noVerticalAlign={false}
    size="sm"
  />
</button>
`;

exports[`menu toggle shows isPrimary 1`] = `
<button
  aria-expanded={false}
  className="pf-c-menu-toggle pf-m-primary"
  type="button"
>
  <span
    className="pf-c-menu-toggle__text"
  >
    Toggle
  </span>
  <span
    className="pf-c-menu-toggle__controls"
  >
    <span
      className="pf-c-menu-toggle__toggle-icon"
    >
      <CaretDownIcon
        aria-hidden={true}
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
    </span>
  </span>
</button>
`;
